go-controller/pkg/ovn/pods.go

```diff
@@ -172,10 +173,7 @@ func (oc *Controller) addRoutesGatewayIP(pod *kapi.Pod, podAnnotation *util.PodA
 	// DUALSTACK FIXME: hybridOverlayExternalGW is not Dualstack
 	var hybridOverlayExternalGW net.IP
 	if config.HybridOverlay.Enabled {
-		hybridOverlayExternalGW, err = oc.getHybridOverlayExternalGwAnnotation(pod.Namespace)
-		if err != nil {
-			return err
-		}
+		hybridOverlayExternalGW = net.ParseIP(pod.Annotations[hotypes.HybridOverlayExternalGw])
```

Comment:
this only works on pod update, and at pod create time we need the right EXGW route